### ISSUE TITLE
Find LLM automatically

### ISSUE THREAD

**Issue by SubatomicPlanets** [undreamai/LLMUnity#173]
### Describe the feature

Right now you need to select the LLM component in the LLMCharacter inspector. If the LLMCharacter is in a prefab or another scene, this won't work. You can assign it in a script using llmcharacter.llm = llm, but there may be better ways. 

You could for instance find the first LLM component in the scene in the awake method of the LLMCharacter and use that. Or you could use a singleton. I am not sure if you can have multiple LLM components right now. If not, then I think something like a singleton would be nice, since that is also used in official  Unity packages like netcode for gameobjects (where you have a singleton on one NetworkManager, and multiple NetworkObjects which use the singleton). If you can have multiple LLM components then I would still keep a setting in the inspector, but if it is set to null, then it should just use the first LLM it finds.

I am not sure if this is needed, but I think it makes sense. Thanks!

**Comment by amakropoulos** [comment]
hi! Yes, you can have multiple LLM objects in parallel.
One way I'm thinking about this is to automatically set the first LLM but only when in the inspector.
If someone creates the LLMCharacter programmatically they would need to set the LLM.
what do you think?

p.s. thanks for the different suggestions! If you like you can also join the [discord community](https://discord.gg/RwXKQb6zdv).

**Comment by SubatomicPlanets** [comment]
Okay, since you can have multiple LLM components this complicates it a bit. My specific use case is two scenes. The first includes the LLM while the second includes the LLMCharacter. I obviously can't set its llm value since the LLM is in a different scene. So I would need to make a script which finds the first LLM and sets the llm value in the Awake method of the character. I tought that this could just be done in the actual LLMCharacter code. If the LLMCharacter spawns but it's llm value is null, it will try and find an LLM and if it exists then it should use that. You could still control it in the inspector or a script. I don't know if this would work properly and be intuitive though... maybe it could also log "No llm provided, using first" or something like that. This is very low priority anyway, since you can achieve this already by doing it manually, and you probably already have a script on the character for actually chatting to it, so it is as easy as adding a line of code. Thanks.

**Comment by amakropoulos** [comment]
I implemented a LLM selector (drop-down) that uses the first LLM found if null here: #182
I don't know if it will work on a different scene though

**Comment by amakropoulos** [comment]
I did another implementation and works across scenes as well
